includes/authentication.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Authentication functionality
+ *
+ * @package  10up-experience
+ */
+
+namespace tenup;
+
+/**
+ * Prevent users from authenticating if they are using a weak password
+ *
+ * @param WP_User $user User object
+ * @param string  $username Username
+ * @param string  $password Password
+ *
+ * @return \WP_User|\WP_Error
+ */
+function prevent_weak_password_auth( $user, $username, $password ) {
+	$test_tlds = array( 'test', 'dev', 'local', '' );
+	$tld       = preg_replace( '#^.*\.(.*)$#', '$1', wp_parse_url( site_url(), PHP_URL_HOST ) );
+
+	if ( ! in_array( $tld, $test_tlds, true ) && in_array( strtolower( trim( $password ) ), weak_passwords(), true ) ) {
+		return new \WP_Error( 'Auth Error', sprintf( '%s <a href="%s">%s</a> %s',
+			__( 'Please', 'tenup' ),
+			esc_url( wp_lostpassword_url() ),
+			__( 'reset your password', 'tenup' ),
+			__( 'in order to meet recently improved security measures.', 'tenup' )
+		) );
+	}
+
+	return $user;
+}
+
+add_filter( 'authenticate', __NAMESPACE__ . '\prevent_weak_password_auth', 30, 3 );
+
+/**
+ * List of the most popular weak passwords
+ *
+ * @return array
+ */
+function weak_passwords() {
```

Comment:
I'm not really excited about having a list of (bad) passwords in here like this and including profanity in something deployed to client sites. Something like zxcvbn might be overkill on the other end of the spectrum, but maybe this should at least be hashed?

Reply:
@ivanlopez Could you let me know where you got this list from?

Reply:
@helen it was from this article about the most popular passwords. http://fortune.com/2017/12/19/the-25-most-used-hackable-passwords-2017-star-wars-freedom/